### PR TITLE
.github: Complete migration from `CODE_OWNERS.txt`

### PR DESCRIPTION
**Pull request by AnthonyLatsis**
* @akyrtzi Do you want me to assign you to IDE, Index, SourceKit, and swift-ide-test files per the old `CODEOWNERS.txt`?
* @belkadan Do you by chance know who took charge of serialization & frontend files? Do you want to receive notifications about PRs that modify these files?
* @DougGregor @eeckstein @nkcsgexi @rjmccall @jckarter @slavapestov I migrated the `CODEOWNERS.txt` info associated with you to the new format. This means that you will start receiving notifications about PRs that modify the files you own per the old specification. Please let me know if I missed anything, the old specification is outdated, or you do not want to receive notifications.
* @jckarter Is you being a code owner of "everything in Swift not covered by someone else" still accurate? If we are going to add a mapping for this, I propose to keep it as a comment until we reach a decent coverage rate, or else your mailbox might go ballistic 🙂.
* @adrian-prantl @artemcm @tshortli @compnerd Assigned you to a few more places you seem to have missed based on existing mappings.

### CODE REVIEW

**Comment by tshortli** [comment]
> Do you by chance know who took charge of Serialization & Frontend files?

Serialization would be @xymus. I'll ask around about `Frontend`.

**Comment by akyrtzi** [comment]
> @akyrtzi Do you want me to assign you to IDE, Index, SourceKit, and swift-ide-test files per the old CODEOWNERS.txt?

No, it should be @bnbarham

**Comment by tshortli** [comment]
@artemcm and I can be listed for `Frontend`.

**Review by adrian-prantl — APPROVED** [review]

File: .github/CODEOWNERS
Comment (AnthonyLatsis) [inline .github/CODEOWNERS:1]: @xymus Mapped you to serialization files. This means that you will start receiving notifications about PRs that modify these files. Please let me know if I missed anything or you if do not want to receive notifications.

**Comment by AnthonyLatsis** [comment]
All right, I have resurrected `CODE_OWNERS.TXT` to unblock this. We will delete it once an updated code ownership governance lands on the website.

**Comment by AnthonyLatsis** [comment]
@swift-ci please smoke test

**Pull request merged**
